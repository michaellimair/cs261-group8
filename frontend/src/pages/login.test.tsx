import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import LoginPage from './login';

describe('login', () => {
  it('renders properly', () => {
    const result = render(
      <BrowserRouter>
        <LoginPage />
      </BrowserRouter>,
    );
    expect(result).toMatchSnapshot();
  });

  it('goes to register page properly', async () => {
    const result = render(
      <BrowserRouter>
        <LoginPage />
      </BrowserRouter>,
    );
    const registerButton = result.queryByTestId('register-button');
    if (!registerButton) {
      throw new Error('Register button cannot be found!');
    }
    registerButton.click();
    expect(window.location.pathname).toBe('/auth/register');
  });
});
