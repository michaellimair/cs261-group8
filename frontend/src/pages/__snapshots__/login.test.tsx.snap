// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="chakra-stack css-owjkmg"
      >
        <h2
          class="chakra-heading css-1gjuw9l"
        >
          login
        </h2>
      </div>
      <div
        class="css-3hziw7"
      >
        <div
          class="chakra-stack css-17kdbol"
        >
          <form
            data-testid="loginForm"
          >
            <div
              class="chakra-form-control css-wlea3r"
              data-testid="username-formcontrol"
              role="group"
            >
              <label
                class="chakra-form__label css-uvho5g"
                data-testid="username-label"
                for="username"
                id="username-label"
              >
                username
                <span
                  aria-hidden="true"
                  class="chakra-form__required-indicator css-0"
                  role="presentation"
                >
                  *
                </span>
              </label>
              <div
                class="chakra-input__group css-4302v8"
              >
                <input
                  aria-required="true"
                  autocomplete="username"
                  class="chakra-input css-0"
                  data-testid="username"
                  id="username"
                  name="username"
                  required=""
                  type="text"
                />
              </div>
            </div>
            <div
              class="chakra-form-control css-wlea3r"
              data-testid="password-formcontrol"
              role="group"
            >
              <label
                class="chakra-form__label css-uvho5g"
                data-testid="password-label"
                for="password"
                id="password-label"
              >
                password
                <span
                  aria-hidden="true"
                  class="chakra-form__required-indicator css-0"
                  role="presentation"
                >
                  *
                </span>
              </label>
              <div
                class="chakra-input__group css-4302v8"
              >
                <input
                  aria-required="true"
                  autocomplete="password"
                  class="chakra-input css-0"
                  data-testid="password"
                  id="password"
                  name="password"
                  required=""
                  type="password"
                />
                <div
                  class="chakra-input__right-element css-1aezdzf"
                  data-testid="show-password-toggle"
                >
                  <button
                    class="chakra-button css-5c2176"
                    data-testid="show-password-button"
                    type="button"
                  >
                    <svg
                      class="chakra-icon css-onkibi"
                      data-testid="password-hidden"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <g
                        fill="currentColor"
                      >
                        <path
                          d="M23.2,10.549a20.954,20.954,0,0,0-4.3-3.6l4-3.995a1,1,0,1,0-1.414-1.414l-.018.018a.737.737,0,0,1-.173.291l-19.5,19.5c-.008.007-.018.009-.026.017a1,1,0,0,0,1.631,1.088l4.146-4.146a11.26,11.26,0,0,0,4.31.939h.3c4.256,0,8.489-2.984,11.051-5.8A2.171,2.171,0,0,0,23.2,10.549ZM16.313,13.27a4.581,4.581,0,0,1-3,3.028,4.3,4.3,0,0,1-3.1-.19.253.253,0,0,1-.068-.407l5.56-5.559a.252.252,0,0,1,.407.067A4.3,4.3,0,0,1,16.313,13.27Z"
                        />
                        <path
                          d="M7.615,13.4a.244.244,0,0,0,.061-.24A4.315,4.315,0,0,1,7.5,12,4.5,4.5,0,0,1,12,7.5a4.276,4.276,0,0,1,1.16.173.244.244,0,0,0,.24-.062l1.941-1.942a.254.254,0,0,0-.1-.421A10.413,10.413,0,0,0,12,4.75C7.7,4.692,3.4,7.7.813,10.549a2.15,2.15,0,0,0-.007,2.9,21.209,21.209,0,0,0,3.438,3.03.256.256,0,0,0,.326-.029Z"
                        />
                      </g>
                    </svg>
                  </button>
                </div>
              </div>
            </div>
            <div
              class="chakra-stack css-1lyzlj9"
            >
              <div
                class="chakra-stack css-1bnf7oa"
              >
                <label
                  class="chakra-checkbox css-1uiwwan"
                >
                  <input
                    class="chakra-checkbox__input"
                    style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                    type="checkbox"
                    value=""
                  />
                  <span
                    aria-hidden="true"
                    class="chakra-checkbox__control css-dnty2r"
                  />
                  <span
                    class="chakra-checkbox__label css-1oeb2oe"
                  >
                    remember_me
                  </span>
                </label>
                <a
                  class="chakra-link css-p03q1r"
                  href="/forgot-password"
                >
                  forgot_password
                </a>
              </div>
              <button
                class="chakra-button css-18dry65"
                data-testid="loginButton"
                type="submit"
              >
                login
              </button>
              <div
                class="chakra-form-control css-t9xfd"
                role="group"
              />
            </div>
            <div
              class="chakra-stack css-nkdhe4"
            >
              <p
                class="chakra-text css-xi606m"
              >
                no_account
                 
                <a
                  class="chakra-link css-p03q1r"
                  data-testid="register-button"
                  href="/auth/register"
                >
                  register
                </a>
              </p>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="chakra-stack css-owjkmg"
    >
      <h2
        class="chakra-heading css-1gjuw9l"
      >
        login
      </h2>
    </div>
    <div
      class="css-3hziw7"
    >
      <div
        class="chakra-stack css-17kdbol"
      >
        <form
          data-testid="loginForm"
        >
          <div
            class="chakra-form-control css-wlea3r"
            data-testid="username-formcontrol"
            role="group"
          >
            <label
              class="chakra-form__label css-uvho5g"
              data-testid="username-label"
              for="username"
              id="username-label"
            >
              username
              <span
                aria-hidden="true"
                class="chakra-form__required-indicator css-0"
                role="presentation"
              >
                *
              </span>
            </label>
            <div
              class="chakra-input__group css-4302v8"
            >
              <input
                aria-required="true"
                autocomplete="username"
                class="chakra-input css-0"
                data-testid="username"
                id="username"
                name="username"
                required=""
                type="text"
              />
            </div>
          </div>
          <div
            class="chakra-form-control css-wlea3r"
            data-testid="password-formcontrol"
            role="group"
          >
            <label
              class="chakra-form__label css-uvho5g"
              data-testid="password-label"
              for="password"
              id="password-label"
            >
              password
              <span
                aria-hidden="true"
                class="chakra-form__required-indicator css-0"
                role="presentation"
              >
                *
              </span>
            </label>
            <div
              class="chakra-input__group css-4302v8"
            >
              <input
                aria-required="true"
                autocomplete="password"
                class="chakra-input css-0"
                data-testid="password"
                id="password"
                name="password"
                required=""
                type="password"
              />
              <div
                class="chakra-input__right-element css-1aezdzf"
                data-testid="show-password-toggle"
              >
                <button
                  class="chakra-button css-5c2176"
                  data-testid="show-password-button"
                  type="button"
                >
                  <svg
                    class="chakra-icon css-onkibi"
                    data-testid="password-hidden"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <g
                      fill="currentColor"
                    >
                      <path
                        d="M23.2,10.549a20.954,20.954,0,0,0-4.3-3.6l4-3.995a1,1,0,1,0-1.414-1.414l-.018.018a.737.737,0,0,1-.173.291l-19.5,19.5c-.008.007-.018.009-.026.017a1,1,0,0,0,1.631,1.088l4.146-4.146a11.26,11.26,0,0,0,4.31.939h.3c4.256,0,8.489-2.984,11.051-5.8A2.171,2.171,0,0,0,23.2,10.549ZM16.313,13.27a4.581,4.581,0,0,1-3,3.028,4.3,4.3,0,0,1-3.1-.19.253.253,0,0,1-.068-.407l5.56-5.559a.252.252,0,0,1,.407.067A4.3,4.3,0,0,1,16.313,13.27Z"
                      />
                      <path
                        d="M7.615,13.4a.244.244,0,0,0,.061-.24A4.315,4.315,0,0,1,7.5,12,4.5,4.5,0,0,1,12,7.5a4.276,4.276,0,0,1,1.16.173.244.244,0,0,0,.24-.062l1.941-1.942a.254.254,0,0,0-.1-.421A10.413,10.413,0,0,0,12,4.75C7.7,4.692,3.4,7.7.813,10.549a2.15,2.15,0,0,0-.007,2.9,21.209,21.209,0,0,0,3.438,3.03.256.256,0,0,0,.326-.029Z"
                      />
                    </g>
                  </svg>
                </button>
              </div>
            </div>
          </div>
          <div
            class="chakra-stack css-1lyzlj9"
          >
            <div
              class="chakra-stack css-1bnf7oa"
            >
              <label
                class="chakra-checkbox css-1uiwwan"
              >
                <input
                  class="chakra-checkbox__input"
                  style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                  type="checkbox"
                  value=""
                />
                <span
                  aria-hidden="true"
                  class="chakra-checkbox__control css-dnty2r"
                />
                <span
                  class="chakra-checkbox__label css-1oeb2oe"
                >
                  remember_me
                </span>
              </label>
              <a
                class="chakra-link css-p03q1r"
                href="/forgot-password"
              >
                forgot_password
              </a>
            </div>
            <button
              class="chakra-button css-18dry65"
              data-testid="loginButton"
              type="submit"
            >
              login
            </button>
            <div
              class="chakra-form-control css-t9xfd"
              role="group"
            />
          </div>
          <div
            class="chakra-stack css-nkdhe4"
          >
            <p
              class="chakra-text css-xi606m"
            >
              no_account
               
              <a
                class="chakra-link css-p03q1r"
                data-testid="register-button"
                href="/auth/register"
              >
                register
              </a>
            </p>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
